Source/DataModels/Scene/Scene.cpp

```diff
@@ -742,6 +868,9 @@ void Scene::UpdateScenePointLights()
 				pl.color = float4(pointLightComp->GetColor(), pointLightComp->GetIntensity());
 
 				pointLights.push_back(pl);
+				cachedPoints.push_back(std::make_pair(pointLightComp, pos));
```

Comment:
Is this `pos` variable necessary? Couldn't we not do
```suggestion
				cachedPoints.push_back(std::make_pair(pointLightComp, cachedPoints.size()));
```
Same for the other loops.

Reply:
Yeah we could to do it. Actually, it would be cachedPoints.size() - 1, but i prefer having it in a separate variable. I find it more clear for it's purpose and readability, and the complexity is the same as doing a size().